pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -530,111 +530,132 @@ public String getTypeString() {
         return "Null";
     }
 
-    @Override
     public CompletableFuture<AnalyzeBacklogResult> analyzeBacklog(Optional<Position> position) {
+        final ManagedLedger managedLedger = topic.getManagedLedger();
+        try {
```

Comment:
Do we need to try-catch the entire method?

Reply:
> Maybe we don't need to try-catch the entire method.

No, `ManagedLedgerException` is not a runtime exception, it needs to be try-catch at another space if do not try-catch it here.

Reply:
Could we create a param out of try-catch block for nonDurableCursor? Such as this.

```
ManagedCursor newNonDurableCursor = null;
try {
    newNonDurableCursor = managedLedger.newNonDurableCursor(PositionImpl.EARLIEST);
} catch (ManagedLedgerException e) {
    throw new RuntimeException(e);
}
Position currentPosition = newNonDurableCursor.getMarkDeletedPosition();
```

Reply:
Improved